tests/templates/spark-mock-task-runner-encrypted.yaml

```diff
@@ -0,0 +1,76 @@
+apiVersion: "sparkoperator.k8s.io/v1beta2"
+kind: SparkApplication
+metadata:
+  name: {{ .Name }}
+  namespace: {{ .Namespace }}
+spec:
+  type: Scala
+  mode: cluster
+  image: {{ .Image }}
+  imagePullPolicy: Always
+  mainClass: MockTaskRunner
+  mainApplicationFile: "https://infinity-artifacts.s3.amazonaws.com/scale-tests/dcos-spark-scala-tests-assembly-2.4.0-20190325.jar"
+  sparkConf:
+    "spark.scheduler.maxRegisteredResourcesWaitingTime": "2400s"
+    "spark.scheduler.minRegisteredResourcesRatio": "1.0"
+    {{ range $key, $value := .Params.SparkConf }}
+    "{{ $key }}": "{{ $value }}"
+    {{ end }}
+  sparkVersion: {{ .SparkVersion }}
+  arguments:
+  {{ range $i, $arg := index .Params.Args }}
+  - "{{ $arg }}"
+  {{ end }}
+  restartPolicy:
+    type: Never
+  driver:
+    cores: 1
+    memory: "512m"
+    labels:
+      version: {{ .SparkVersion }}
+    serviceAccount: {{ .ServiceAccount }}
+    {{- if .Params.SslEnabled }}
+    secrets:
+      - name: {{ .Params.SslSecrets }}
+        path: "/tmp/spark/ssl"
+        secretType: Generic
+    {{- end }}
+    env:
```

Comment:
can we check, if it is possible to specify the environment in a less verbose way via `sparkConf`? E.g.:
```
sparkConf:
  "spark.kubernetes.driver.secretKeyRef.SPARK_AUTHENTICATE_SECRET": "{{ .Params.SparkSecrets }}:auth-secret"
  "spark.kubernetes.driver.secretKeyRef.SPARK_SSL_KEYPASSWORD": "{{ .Params.SparkSecrets }}:key-password"
  "spark.kubernetes.driver.secretKeyRef.SPARK_SSL_KEYSTOREPASSWORD": "{{ .Params.SparkSecrets }}:keystore-password"
  "spark.kubernetes.driver.secretKeyRef.SPARK_SSL_TRUSTSTOREPASSWORD": "{{ .Params.SparkSecrets }}:truststore-password"
  "spark.kubernetes.executor.secretKeyRef.SPARK_AUTHENTICATE_SECRET": "{{ .Params.SparkSecrets }}:auth-secret"
```